executor/coprocessor.go

```diff
@@ -32,50 +33,93 @@ import (
 
 // CoprocessorDAGHandler uses to handle cop dag request.
 type CoprocessorDAGHandler struct {
-	sctx    sessionctx.Context
-	resp    *coprocessor.Response
-	selResp *tipb.SelectResponse
-	dagReq  *tipb.DAGRequest
+	sctx   sessionctx.Context
+	dagReq *tipb.DAGRequest
 }
 
 // NewCoprocessorDAGHandler creates a new CoprocessorDAGHandler.
 func NewCoprocessorDAGHandler(sctx sessionctx.Context) *CoprocessorDAGHandler {
 	return &CoprocessorDAGHandler{
-		sctx:    sctx,
-		resp:    &coprocessor.Response{},
-		selResp: &tipb.SelectResponse{},
+		sctx: sctx,
 	}
 }
 
 // HandleRequest handles the coprocessor request.
 func (h *CoprocessorDAGHandler) HandleRequest(ctx context.Context, req *coprocessor.Request) *coprocessor.Response {
 	e, err := h.buildDAGExecutor(req)
 	if err != nil {
-		return h.buildResponse(err)
+		return h.buildErrorResponse(err)
 	}
 
 	err = e.Open(ctx)
 	if err != nil {
-		return h.buildResponse(err)
+		return h.buildErrorResponse(err)
 	}
 
 	chk := newFirstChunk(e)
 	tps := e.base().retFieldTypes
+	var totalChunks, partChunks []tipb.Chunk
 	for {
 		chk.Reset()
 		err = Next(ctx, e, chk)
 		if err != nil {
-			break
+			return h.buildErrorResponse(err)
 		}
 		if chk.NumRows() == 0 {
 			break
 		}
-		err = h.appendChunk(chk, tps)
+		partChunks, err = h.buildChunk(chk, tps)
 		if err != nil {
+			return h.buildErrorResponse(err)
+		}
+		totalChunks = append(totalChunks, partChunks...)
+	}
+	return h.buildResponse(totalChunks)
+}
+
+// HandleStreamRequest handles the coprocessor stream request.
+func (h *CoprocessorDAGHandler) HandleStreamRequest(ctx context.Context, req *coprocessor.Request, stream tikvpb.Tikv_CoprocessorStreamServer) error {
+	e, err := h.buildDAGExecutor(req)
+	if err != nil {
+		return stream.Send(h.buildErrorResponse(err))
+	}
+
+	err = e.Open(ctx)
+	if err != nil {
+		return stream.Send(h.buildErrorResponse(err))
+	}
+
+	chk := newFirstChunk(e)
+	tps := e.base().retFieldTypes
+	for {
+		chk.Reset()
+		if err = Next(ctx, e, chk); err != nil {
+			return stream.Send(h.buildErrorResponse(err))
+		}
+		if chk.NumRows() == 0 {
 			break
 		}
+		err = h.buildResponseAndSendToStream(chk, tps, stream)
+		if err != nil {
+			return stream.Send(h.buildErrorResponse(err))
+		}
 	}
-	return h.buildResponse(err)
+	return h.buildResponseAndSendToStream(chk, tps, stream)
```

Comment:
This line can move into line 100, and use return.

Reply:
done.